File: scripts/past_problems/4.rb
A = gets.to_i
B = gets.to_i
C = gets.to_i
X = gets.to_i

res = 0

(0..A).each do |a|
  (0..B).each do |b|
    (0..C).each do |c|
      total = 500 * a + 100 * b + 50 * c
      res += 1 if total == X
    end
  end
end

puts res

File: scripts/past_problems/4.cpp
// 解き直しが必要。
#include <iostream>
using namespace std;

int main() {
  int A, B, C, X;
  cin >> A >> B >> C >> X;
  int res = 0;
  for (int a = 0; a <= A; ++a) {
    for (int b = 0; b <= B; ++b) {
      for (int c = 0; c <= C; ++c) {
        int total = 500*a + 100*b + 50*c; // 合計金額
        if (total == X) ++res; // 合計金額が X に一致したら答えをインクリメント
      }
    }
  }
  cout << res << endl;
}

